	
syntax = "proto3";

message MetricRecord {

    message Timeseriesdescriptor {
        string container_name = 1;
        string project_id = 2;
        string location = 3;
        string cluster_name = 4;
        string namespace_name = 5;
        string controller_name = 6;
        string controller_type = 7;
    }

    message Timeinterval {
        string start_time = 1;
        string end_time = 2;
    }

    message Values {
        int64 int64Value = 1;
        double doubleValue = 2;
    }

    message Pointdata {
        Timeinterval timeInterval = 1;
        Values values = 2;
    }

    Timeseriesdescriptor timeSeriesDescriptor = 1;
    Pointdata pointData = 2;
    string metricName = 3;
}